packages/babel-plugin-proposal-class-properties/test/fixtures/private/optional-chain/output.js

```diff
@@ -0,0 +1,43 @@
+function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
+
+function _classStaticPrivateFieldSpecGet(receiver, classConstructor, descriptor) { if (receiver !== classConstructor) { throw new TypeError("Private static access of wrong provenance"); } if (descriptor.get) { return descriptor.get.call(receiver); } return descriptor.value; }
+
+class Foo {
+  static test() {
+    var _o, _ref, _o2, _ref2, _o3, _ref3, _ref4, _ref5, _ref6, _ref7, _ref8, _ref9, _o4, _ref10, _o5, _ref11, _o6, _ref12, _ref13, _ref14, _o7, _ref15, _ref16, _ref17, _o8, _ref18, _ref19, _ref20;
+
+    const o = {
+      Foo: Foo
+    };
+    const deep = {
+      very: {
+        o
+      }
+    };
+    (_ref = (_o = o)?.Foo, _o === void 0 || _o === null) ? void 0 : _classStaticPrivateFieldSpecGet(_ref, Foo, _x);
```

Comment:
Can we produce this simpler output for `o?.Foo.#x`?

```js
(_o = o) ?? _classStaticPrivateFieldSpecGet(_o.Foo, Foo, _x);
```

Reply:
Unfortunately not, since that would only execute `_classStaticPrivateFieldSpecGet`'s branch when `o` was nullish! Then `_o.Foo` would be guaranteed to throw.

We'd need an inverse operator, something like "[not-nullish and](https://github.com/tc39/proposal-nullish-coalescing/issues/61)".

Reply:
Oh right :man_facepalming: 

I think that we can still simplify it a bit, by avoiding the double nullish check (the manual one and the one in `?.`) and a temporary variable:
```js
(_o = o) === null || _o === void 0 ? void 0 : _classStaticPrivateFieldSpecGet(_o.Foo, Foo, _x)
```

(Please put the `null` check first, to that `=== void 0 ? void 0 :` matches the existing optional chaining transform and can be gzipped more)

Reply:
Yah... why didn't I think of that?